docs/case-control-studies.md

```diff
@@ -0,0 +1,150 @@
+There are five steps to undertaking a [case-control study](https://en.wikipedia.org/wiki/Case%E2%80%93control_study) with OpenSAFELY:
+
+1. Extract data for the cases
+2. Extract data for the potential controls
+3. Match the cases to the potential controls
+4. Extract data for the matched controls
+5. Analysis
+
+To begin with, our [*project.yaml*](actions-pipelines.md) looks like this:
+
+```yaml
+version: '3.0'
+
+expectations:
+  population_size: 1000
+
+actions:
+  # Extract data for the cases
+  # Extract data for the potential controls
+  # Match the cases to the potential controls
+  # Extract data for the matched controls
+```
+
+## Extract data for the cases
+
+In this step, we will construct [a study definition](study-def.md) to extract all the data we need for the cases:
+that is, the *matching data*, or the data we will use to match the cases to the potential controls;
+and the *non-matching data*, or the data we will use for analysis.
+To avoid duplicating code to extract the matching data and the non-matching data in this and the following steps,
+we could use separate Python scripts to [share common study definition variables](study-def-tricks.md#sharing-common-study-definition-variables).
+
+As we will construct multiple study definitions in this and the following steps, we will name this study definition *study_definition_cases.py*.
+When working with [multiple study definitions](study-def.md#multiple-study-definitions), each study definition's suffix is used to name each corresponding output file.
+Here, the study definition's suffix is *cases*, so the corresponding output file will be named *input_cases.csv*.
+
+Our *project.yaml* now includes the following action:
+
+```yaml
+# ...
+actions:
+  extract_cases:
+    run: cohortextractor:latest generate_cohort --study-definition study_definition_cases
+    outputs:
+      highly_sensitive:
+        cohort: output/input_cases.csv
+```
+
+## Extract data for the potential controls
+
+The *potential controls* are the group of patients that are matched to the cases to give the *matched controls*.
+In this step, we will construct a second study definition to extract only the matching data for the potential controls.
+We will name this study definition *study_definition_potential_controls.py*.
+
+
+Our *project.yaml* now includes the following action:
+
+```yaml
+# ...
+actions:
+  # ...
+  extract_potential_controls:
+    run: cohortextractor:latest generate_cohort --study-definition study_definition_potential_controls
+    outputs:
+      highly_sensitive:
+        cohort: output/input_potential_controls.csv
+```
+
+## Match the cases to the potential controls
+
+In this step, we will use the [OpenSAFELY matching library](https://github.com/opensafely-core/matching#readme) in [a scripted action](actions-scripts.md) to match the cases to the potential controls.
+We will name this scripted action *match.py*.
```

Comment:
Could add "You can also use your own custom action to match cases to controls."

Reply:
Thanks @alexwalkerepi 🙂 Done in 4a2b6ba.